eventuate-messaging-kafka-basic-consumer/src/main/java/io/eventuate/messaging/kafka/basic/consumer/KafkaMessageConsumer.java

```diff
@@ -0,0 +1,33 @@
+package io.eventuate.messaging.kafka.basic.consumer;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Set;
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+
+public interface KafkaMessageConsumer extends KafkaConsumerOffsetStorageStrategy {
```

Comment:
`KafkaMessageConsumer` should only define methods that are used by the existing code to consume messages: think interface separation principle.

It does not need to extend `KafkaConsumerOffsetStorageStrategy` since that functionality is specific to a custom implementation.